raster-foundry/raster-foundry
# Remove complex color correction

## Overview

This PR removes users' ability to switch color modes and adjust color correction parameters. Hardly anyone ever used it and it made code complicated, so now it's no longer present.

### Checklist


### Demo

![image](https://user-images.githubusercontent.com/5702984/68976899-939ce300-07bc-11ea-99ec-6c85d9185c21.png)

with no errors :tada:

### Notes

`ColorCorrect.scala` is still around with slightly cleaned up names of things. Removing it entirely turned out to be a headache and to reintroduce nodata bugs, so instead, it still lives, and its complicated logic has been removed. V2 UI still has color correction, since it's not super important to spend time removing code that exposes features that no one ever had access to. It's still technically possible to persistently update color corrections via the API, and you can still specify other bands with the `redBand=x&greenBand=y&blueBand=z` parameters.  I didn't write a migration to drop the color correction column, since that persistence keeps `mosaicDefinitionToImage` pure, which saves us a db round trip.

## Testing Instructions

- confirm the frontend builds without errors
- look at project scenes and note that everything is fine
- make some tile requests with the band parameters and confirm you can still re-color things if you really want to

Closes #5239
